src/Compilers/Core/Portable/SignatureComparer.cs

```diff
@@ -111,26 +118,30 @@ public bool MatchMethodSignature(MethodSymbol method, ImmutableArray<byte> signa
         }
 
         private bool MatchParameter(ParameterSymbol parameter, ImmutableArray<byte> signature, ref int position)
+        {
+            bool isByRef = IsByRef(signature, ref position);
+
+            if (IsByRefParam(parameter) != isByRef)
+            {
+                return false;
+            }
+
+            return MatchType(GetParamType(parameter), signature, ref position);
+        }
+
+        private static bool IsByRef(ImmutableArray<byte> signature, ref int position)
```

Comment:
Seems like this would be better as a local function. It modifies parent state, isn't particularly reusable, and is only used in one place.

Reply:
It looks like it's used in two places

---
In reply to: [153348272](https://github.com/dotnet/roslyn/pull/23363#discussion_r153348272) [](ancestors = 153348272)

Reply:
yes, it is used in two places

---
In reply to: [153592425](https://github.com/dotnet/roslyn/pull/23363#discussion_r153592425) [](ancestors = 153592425,153348272)